### PR TITLE
feat(sdk): enable read in component using IR

### PR DESCRIPTION
**Pull request by connor-mccarthy**
**Description of your changes:**
Enables reading in a primitive component definition from IR YAML.

**Checklist:**
- [x] The title for your pull 
request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: sdk/python/kfp/components/structures.py [hunk 1]
```diff
@@ -154,8 +270,8 @@ class InputUriPlaceholder(base_model.BaseModel, PlaceholderSerializationMixin):
     _FROM_PLACEHOLDER_REGEX = r"\{\{\$\.inputs\.artifacts\[(?:''|'|\")(.+?)(?:''|'|\")]\.uri\}\}"
 
 
-class OutputParameterPlaceholder(base_model.BaseModel,
-                                 PlaceholderSerializationMixin):
+class OutputValuePlaceholder(base_model.BaseModel,
```
Comment (chensun) [inline sdk/python/kfp/components/structures.py:273]: `OutputParameterPlaceholder` is probably a better name. We don't have a `{outputValue: }` placeholder per se. The placeholder is still an **output (file) path** (as indicated by `.output_file`) for parameter.

Comment (connor-mccarthy) [reply sdk/python/kfp/components/structures.py]: Thanks, @chensun. Updated!

**Review by chensun — COMMENTED** [review]
/lgtm
/approve

Thanks!

**Pull request merged**